File: api-legacy/src/main/java/com/bitstachio/bazaarapi/user/controller/UserController.java
package com.bitstachio.bazaarapi.user.controller;

import com.bitstachio.bazaarapi.user.dto.UserCreateRequest;
import com.bitstachio.bazaarapi.user.dto.UserResponse;
import com.bitstachio.bazaarapi.user.dto.UserUpdateRequest;
import com.bitstachio.bazaarapi.user.service.UserService;

import lombok.RequiredArgsConstructor;

import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.UUID;

@RestController
@RequestMapping("/api/users")
@RequiredArgsConstructor
public class UserController {

    private final UserService userService;

    @PostMapping
    public UserResponse create(@RequestBody UserCreateRequest request) {
        return userService.create(request);
    }

    @GetMapping("/{id}")
    public UserResponse getById(@PathVariable UUID id) {
        return userService.getById(id);
    }

    @GetMapping
    public List<UserResponse> getAll() {
        return userService.getAll();
    }

    @PutMapping("/{id}")
    public UserResponse update(@PathVariable UUID id, @RequestBody UserUpdateRequest request) {
        return userService.update(id, request);
    }

    @DeleteMapping("/{id}")
    public void delete(@PathVariable UUID id) {
        userService.delete(id);
    }
}

File: api/src/main/kotlin/dev/bitstachio/interdimensional_bazaar/user/controller/UserController.kt
package dev.bitstachio.interdimensional_bazaar.user.controller

import dev.bitstachio.interdimensional_bazaar.user.dto.UserCreateRequest
import dev.bitstachio.interdimensional_bazaar.user.dto.UserResponse
import dev.bitstachio.interdimensional_bazaar.user.dto.UserUpdateRequest
import dev.bitstachio.interdimensional_bazaar.user.service.UserService
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.util.UUID

@RestController
@RequestMapping("/api/users")
class UserController(private val userService: UserService) {

	@PostMapping
	fun create(@RequestBody request: UserCreateRequest): UserResponse {
		return userService.create(request)
	}

	@GetMapping("/{id}")
	fun getById(@PathVariable id: UUID): UserResponse {
		return userService.getById(id)
	}

	@GetMapping
	fun getAll(): List<UserResponse> {
		return userService.getAll()
	}

	@PutMapping("/{id}")
	fun update(@PathVariable id: UUID, @RequestBody request: UserUpdateRequest): UserResponse {
		return userService.update(id, request)
	}

	@DeleteMapping("/{id}")
	fun delete(@PathVariable id: UUID) {
		userService.delete(id)
	}
}
